### ISSUE TITLE
Moving Webhook Event implemetations to core.

### ISSUE THREAD

**Issue by criske** [self-xdsd/self-core#926]
I'm testing webhooks for gitlab and noticed that `Github/GithubWebhookEvent` are created in `self-pm` and `self-todos`(well github is created as anon impl and gitlab not yet). Wouldn't be better to have a factory for these and move their implementations here.?
```java
public final class WebhookEvents {
   private WebhookEvents(){}
   public static Event create(String provider, String type, String payload){
        if(provider.equals("github")) new GithubWebhookEvent else ...
   }
 };
}
```
and on `self-pm/self-todos` we will just call something like `Event event = WebhookEvents.create("github", "push", payload)`?

**Comment by charlesmike** [comment]
@criske thank you for reporting this. I'll assign someone to take care of it soon.

**Comment by charlesmike** [comment]
@amihaiemil this is your task now, please go ahead. Deadline (when this ticket should be closed) is ``2021-02-01T13:16:13.672516``.

Estimation here is ``60 minutes``, that's how much you will be paid.

Remember, you don't have to solve everything in this ticket - you can solve it [partially](https://docs.self-xdsd.com/microtasking.html) and leave ``todo`` markers in the code, which will become future tasks.

If you have any questions don't ask me, I'm not a technical person. Open new tickets instead.

If you don't have time or simply don't want to work on this, you can always [resign](https://docs.self-xdsd.com/projectmanager.html#commands).

**Comment by amihaiemil** [comment]
@criske we could, but maybe later. Btw, I working on GitlabWevhookEvent at the moment :D

**Comment by amihaiemil** [comment]
@charlesmike remove

**Comment by charlesmike** [comment]
> @charlesmike remove

@amihaiemil ok, I've removed this task from scope. I'm not managing it anymore.

**Comment by criske** [comment]
@amihaiemil yep I've noticed :D 
Btw, I've discovered a corner case.  Dunno if this applies to github too, but if you push a branch into master and this branch has the same name as a previous merged one that was deleted, "commits" will be empty and thus no latest commit to extract.

**Comment by amihaiemil** [comment]
@charlesmike add

**Comment by charlesmike** [comment]
> @charlesmike add

@amihaiemil I've just registered this ticket as a task and will assign it to someone soon. Thanks!

**Comment by charlesmike** [comment]
@amihaiemil this is your task now, please go ahead. Deadline (when this ticket should be closed) is ``2021-02-11T12:07:50.860391``.

Estimation here is ``60 minutes``, that's how much you will be paid.

Remember, you don't have to solve everything in this ticket - you can solve it [partially](https://docs.self-xdsd.com/microtasking.html) and leave ``todo`` markers in the code, which will become future tasks.

If you have any questions don't ask me, I'm not a technical person. Open new tickets instead.

If you don't have time or simply don't want to work on this, you can always [resign](https://docs.self-xdsd.com/projectmanager.html#commands).

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).
